### ISSUE TITLE
NuGet: Strange Namespace and no *.lib generated

### ISSUE THREAD

**Issue by ArgusMagnus** [3F/DllExport#33]
Hi

I have installed the NuGet package (v1.5.2) to an .NET 4.6.1 library Project. There are two issues:

1. The DllExportAttribute resides in a very strange Namespace: "D3F00FF1770DED978EC774BA389F2DC901F4.B00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.C00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.D00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.E00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.F00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"

2. When I build the Project (x64) no *.lib is generated which can be linked by my C++ lib

**Comment by OmegaExtern** [comment]
You have to unload the project, and open it again (or simply restart Visual Studio).
This is a known issue, and has been discussed previously. Also see #14.

**Comment by 3F** [comment]
@ArgusMagnus 

1. ddNS feature that should update this sequence after installation.
2. You should define property `$(DllExportGenExpLib = true)` to activate this feature because it disabled by default.

Probably you have some errors when install our package, because you should see this:

![](https://raw.githubusercontent.com/3F/DllExport/master/Resources/img/DllExport.png)

Or you should define this manually like:

```xml
<DllExportNamespace>System.Runtime.InteropServices</DllExportNamespace>
<DllExportDDNSCecil>true</DllExportDDNSCecil>
<DllExportOrdinalsBase>1</DllExportOrdinalsBase>
<DllExportGenExpLib>true</DllExportGenExpLib>
<DllExportOurILAsm>false</DllExportOurILAsm>
...
```

Thus, for clarify please provide detailed log when you install this - for VS IDE:

* `Tools` - `Options` - `Project and Solutions` - `Build and Run` - MSBuild project build verbosity

For msbuild.exe:

* `/v:diag`

**Comment by ArgusMagnus** [comment]
That Dialog never showed up.

[msbuild_log.txt](https://github.com/3F/DllExport/files/992028/msbuild_log.txt)

**Comment by 3F** [comment]
> That Dialog never showed up.

seems you have problem with installing, thus please also provide **log when you install this package** (see from Output).

Current package uses init.ps1 & install.ps1 scripts by nuget

It can be related for #36, but I haven't seen any similar problem for both new VS versions: 
* VisualStudio.15.Release/15.2.0+26430.4 
* VisualStudio.15.Preview/15.3.0-Preview+26507.0

that is, the install.ps1 has been obsolete already for VS 14.0 (at least, as I remember) and recommended `\build\...` targets instead (as for my other packages e.g.: `\build\native\ or \build\net\`).

> issue is fixed by adding the above elements ... <Import Project="...

yes, this line should be automatically added after `install.ps1` or `\build\xxx\xxx.targets`. This is required for DllExport anyway.

> Cannot find lib.exe 

The new VS 2017 contains [major changes with its environment](https://plus.google.com/101239554716569212042/posts/1K5tdji63TM) (+[hMSBuild](https://plus.google.com/101239554716569212042/posts/VxHBvfjDf38)), so I know what's going on <_< thanks for notice.

Temporarily try to redefine this:

`<installationPath>\<Version>\<Edition>\VC\Tools\MSVC\<Version>\bin\<HostTarget>\<Arch>`

for example:

```xml
<DllExportLibToolPath>$(DevEnvDir)..\..\VC\Tools\MSVC\14.11.25301\bin\HostX86\x86</DllExportLibToolPath>
```

**For all:** I've plan to consider this project to the end of this month or early (as soon as possible for my time)

**Comment by ArgusMagnus** [comment]
Sorry for responding so late, I only work once a week on the project I use DllExport.

Package Manager Output:

```
Successfully installed 'DllExport 1.5.2' to MyProject
Executing script file 'C:\Users\automation\.nuget\packages\dllexport\1.5.2\tools\init.ps1'...
The names of some imported commands from the module 'DllExportCmdLets' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
Executing nuget actions took 2.51 sec
Time Elapsed: 00:00:02.7435282
========== Finished ==========
```

**Comment by 3F** [comment]
@ArgusMagnus 

yes, yes, only init.ps1 that related for #36
thanks for information

by the way, it still works for me (VisualStudio.15.Release/15.2.0+26430.6):

```
....
Executing script file 'D:\tmp\2017\ConsoleApp1\packages\DllExport.1.5.2\tools\init.ps1'...
The names of some imported commands from the module 'DllExportCmdLets' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
Executing script file 'D:\tmp\2017\ConsoleApp1\packages\DllExport.1.5.2\tools\install.ps1'...
set new namespace(Cecil: True): (ConsoleApp1) - (D:\tmp\2017\ConsoleApp1\packages\DllExport.1.5.2\lib\net20\DllExport.dll)
....
```

so, what your build version of VS15 for clarify ? (Help - About - first 3 lines)

**Comment by ArgusMagnus** [comment]
Microsoft Visual Studio Enterprise 2017 
Version 15.2 (26430.6) Release
VisualStudio.15.Release/15.2.0+26430.6

**Comment by 3F** [comment]
thanks,
anyway I've plan to change this obsolete variant soon :) +PackageReference

for today, please configure this manually (see above) if anyone else have this problem.

**Comment by ArgusMagnus** [comment]
Ok, thanks

**Comment by 3F** [comment]
I closed this issue because of new [Wizard for v1.6+ versions](https://github.com/3F/DllExport/issues/38). You already can test it via our beta releases:

* Latest manager: https://3F.github.io/DllExport/releases/latest/manager/
* Please note again - you shouldn't use standard nuget clients anymore: https://www.youtube.com/watch?v=okPThdWDZMM 

If you still have a some related problems, please create new issue for new planned Wizard.
